app/Http/Controllers/JournalController.php

```diff
@@ -188,4 +192,30 @@ public function deleteEntry(Request $request, $entryId)
         $entry->deleteJournalEntry();
         $entry->delete();
     }
+
+    /**
+     * Display mood trends over time.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function moods(Request $request)
+    {
+        $results = [
+            'unit' => 'months',
+            'range' => 12,
+            'data' => [],
+            'available_units' => ['days', 'weeks', 'months', 'years'],
+        ];
+        $start = Carbon::now()->subMonths($results['range']);
+        $dateIntervalValue = 'P1D';
+        $period = new DatePeriod($start, new DateInterval($dateIntervalValue), Carbon::now());
+        foreach ($period as $date) {
+            $dateString = $date->format('Y-m-d');
+            $rating = DB::table('days')->whereDate('date', $dateString)->pluck('rate')->first();
```

Comment:
1. Why use `DB::table('days')` and not Eloquent?
2. You don't filter by `account_id`, which means you will take all the days for all the accounts here, which is bad 😀

Reply:
Good catch. Will do.